src/main/java/org/spongepowered/api/text/channel/MessageReceiver.java

```
@@ -0,0 +1,86 @@
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.text.channel;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import org.spongepowered.api.text.Text;
+
+/**
+ * Represents something that can receive (and send) messages.
+ */
+public interface MessageReceiver {
+
+    /**
+     * Sends the formatted text message to source when possible. If text formatting
+     * is not supported in the implementation it will be displayed as plain text.
+     *
+     * @param message The message
+     */
+    void sendMessage(Text message);
+
+    /**
+     * Sends the formatted text message(s) to source when possible. If text formatting
+     * is not supported in the implementation it will be displayed as plain text.
+     *
+     * @param messages The message(s)
+     */
+    default void sendMessages(Text... messages) {
+        checkNotNull(messages, "messages");
+        checkArgument(messages.length > 0, "empty array of messages");
```

Yep, It's better that people can't just send `sendMessages();` and expect it to fail silently.